source/Cosmos.System2/FileSystem/Disk.cs

```diff
@@ -0,0 +1,296 @@
+﻿//#define COSMOSDEBUG
+
+using System;
+using System.Collections.Generic;
+using Cosmos.HAL.BlockDevice;
+using Cosmos.System.FileSystem.FAT;
+using Cosmos.System.FileSystem.VFS;
+
+namespace Cosmos.System.FileSystem
+{
+    public class Disk
+    {
+        private List<ManagedPartition> parts = new List<ManagedPartition>();
+        private static List<FileSystemFactory> registeredFileSystems = new List<FileSystemFactory>();
+        public bool IsMBR { get { return !GPT.IsGPTPartition(Host); } }
+        /// <summary>
+        /// The size of the disk in MB.
+        /// </summary>
+        public int Size { get; }
+        /// <summary>
+        /// List of partitions
+        /// </summary>
+        public List<ManagedPartition> Partitions
```

Comment:
Does it make sense to cache the partitions after they have been found once?

Reply:
Not really, as the partitions can change when Add/Remove partitions is called.